flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/window/assigners/CountSlidingWindowAssigner.java

```diff
@@ -67,17 +67,19 @@ public void open(InternalWindowProcessFunction.Context<?, CountWindow> ctx) thro
 		Long countValue = count.value();
 		long currentCount = countValue == null ? 0L : countValue;
 		count.update(currentCount + 1);
-		long lastId = currentCount / windowSlide;
-		long lastStart = lastId * windowSlide;
-		long lastEnd = lastStart + windowSize - 1;
+		long windowId = currentCount / windowSlide;
+		long windowEnd = windowId * windowSlide + windowSlide - 1;
+		long windowStart = windowEnd - windowSize + 1;
 		List<CountWindow> windows = new ArrayList<>();
-		while (lastId >= 0 && lastStart <= currentCount && currentCount <= lastEnd) {
-			if (lastStart <= currentCount && currentCount <= lastEnd) {
-				windows.add(new CountWindow(lastId));
+		while ((windowStart <= currentCount) && (currentCount <= windowEnd)) {
+			if (0 > windowStart) {
+				windows.add(new CountWindow(windowId, windowSize + windowStart));
+			} else {
+				windows.add(new CountWindow(windowId, windowSize));
```

Comment:
What about to encode CountWindow as "countStart, countEnd" like TimeWindow. Then the logic of assigning windows will be similar to `SlidingWindowAssigner` which is much simpler IMO. What do you think?

Reply:
Good idea.